X_mix-tasks.md

```diff
@@ -305,6 +305,76 @@ Compiled web/models/post.ex
 (stdlib) erl_eval.erl:657: :erl_eval.do_apply/6
 ```
 
+If we don't want to create a model for our resource we can use the `--no-model` flag.
+
+```console
+$ mix phoenix.gen.json Post posts title:string content:string --no-model
+* creating web/controllers/post_controller.ex
+* creating web/views/post_view.ex
+* creating test/controllers/post_controller_test.exs
+* creating web/views/changeset_view.ex
+```
+
+It will tell us we need to add a line to our router file, but since we skipped the model, it won't mention anything about `ecto.migrate`.
+
+```console
+Add the resource to the proper scope in web/router.ex:
+
+resources "/posts", PostController
+```
+
+Important: If we don't do this, our application won't compile, and we'll get an error.
+
+```console
+$ mix phoenix.server
+
+== Compilation error on file web/controllers/post_controller.ex ==
+** (CompileError) web/controllers/post_controller.ex:15: HelloPhoenix.Post.__struct__/0 is undefined, cannot expand struct HelloPhoenix.Post
+    (elixir) src/elixir_map.erl:55: :elixir_map.translate_struct/4
+    (stdlib) lists.erl:1352: :lists.mapfoldl/3
+```
+
+#### `mix phoenix.gen.model`
+
+If we don't need a complete HTML/JSON resource and instead are only interested in a model part of it we can use the `phoenix.gen.model` task. It will generate a model, a migration and a test case.
+
+The `phoenix.gen.model` task takes a number of arguments, the module name of the model, the plural model name used for the schema, and a list of column_name:type attributes.
+
+```console
+$ mix phoenix.gen.model User users name:string age:integer
+* creating priv/repo/migrations/20150527185323_create_user.exs
+* creating web/models/user.ex
+* creating test/models/user_test.exs
+```
+
+> Note: If we need to namespace our resource we can simply namespace the first argument of the generator.
+```console
+$ mix phoenix.gen.model Admin.User users name:string age:integer
+* creating priv/repo/migrations/20150527185940_create_admin_user.exs
+* creating web/models/admin/user.ex
+* creating test/models/admin/user_test.exs
+```
+
+#### `mix phoenix.gen.channel`
+
+This task will generate a basic Phoenix channel as well a test case for it. It takes only two arguments, the module name for the channel and plural used as the topic.
+
+```console
+$ mix phoenix.gen.channel Room rooms
+* creating web/channels/room_channel.ex
+* creating test/channels/room_channel_test.exs
+```
+
+When `phoenix.gen.channel` is done creating files, it helpfully tells us that we need to add a channel route to our router file.
```

Comment:
What would you think about removing "creating files"? That would leave:

"When `phoenix.gen.channel` is done, it helpfully . . . "

I think either way is fine, but I thought I would ask.